packages/data-context/src/sources/BrowserDataSource.ts

```diff
@@ -58,6 +72,37 @@ export class BrowserDataSource {
     return this.ctx.coreData.machineBrowsers
   }
 
+  /**
+   * Gets the browsers from the project config, caching the Promise on the
+   * coreData so we only look them up once
+   */
+  userBrowsers () {
+    if (!this.ctx.coreData.userBrowsers) {
+      const p = this.ctx.project.getConfig()
+
+      this.ctx.coreData.userBrowsers = p.then(async (cfg) => {
+        return cfg.browsers?.map<FoundBrowser | undefined>((b) => {
```

Comment:
at this point, won't the `cfg.browsers` include the found machine browsers as well as the user-defined browsers?

Reply:
It does! I didn't catch that. What is the authoritative source for "browsers from the config file" (is this a thing?) vs "browsers added via a plugin" vs "browsers on the machine" vs "resolved browsers from the full config" ?

I'll push a commit that tries to "4" - "3" but it seems suspect.

Reply:
@emilyrohrbough - following up to see if you have feedback on the updates I pushed. Thanks!!